experiments/scripts/run_tune.py

```diff
@@ -32,6 +32,38 @@
 CONFIG_DIR = FILE_PATH.parent / "configs"
 ZERO_RANK = [0, -1]
 
+
+def should_restart(
+    output_dir: str, restart_checkpoint: Union[str, bool]
+) -> Union[str, bool]:
+    """
+    The default behaviour causes an automatic restart from the most recent checkpoint.
+    However, you can also specify
+        - specific checkpoint folders
+        - a value of False to not load a checkpoint
+    """
+    if isinstance(restart_checkpoint, str):
+        # if specific checkpoint provided
+        if os.path.isdir(restart_checkpoint):
+            return restart_checkpoint
+        else:
+            raise ValueError(f"checkpoint cannot be found at {restart_checkpoint}")
+
+    elif isinstance(restart_checkpoint, bool) and restart_checkpoint:
```

Comment:
maybe this could be 
`elif restart_checkpoint is True:`
because restart_checkpoint can only be Union[str, bool] so if it isn't caught by the isinstance(restart_checkpoint, str) it would have to be a bool at this stage?

Reply:
yes agreed